src/index.html

```diff
@@ -3,31 +3,814 @@
+        ]
+      },
+      branded: false
+    })
+    </script>
+    <p>You can define <code>recommendations</code> object on the Exp instance. The object holds the definitions of recommendations. Each key (in example above <code>products</code>) reserves an Array in the instance data. Each recommendation definition must have <code>id</code> and <code>total</code>. Exp will automatically fetch the products and fill in the reserved Array.</p>
+    <p>Recommendation fetching is asynchronous and it is possible that the web-layer will be displayed before the response arrives and before the product list is rendered. To mitigate this we have added two optional settings. <code>loadingKey</code> and <code>deferTrigger</code>.</p>
+    <p>1. <code>loadingKey</code> changes the value of variable defined on isntance to <code>true</code> when the response from recommendations API arrives.</p>
+    <pre><code>var banner = new Exp({
+  context: this,
+  recommendations: {
+    products: {
+      id: "&lt;recommendation id&gt;",
+      total: 4,
+      loadingKey: 'doneLoading'
+    }
+  },
+  data: {
+    doneLoading: false
+  }
+})</code></pre>
+  <p>2. <code>deferTrigger</code> will simply disable banner triggers until the recommendations API returns response.</p>
+    <pre><code>var banner = new Exp({
+  context: this,
+  recommendations: {
+    products: {
+      id: "&lt;recommendation id&gt;",
+      total: 4,
+      deferTrigger: true
+    }
+  },
+  data: {
+    doneLoading: false
+  }
+})</code></pre>
+    <h2 id="event-tracking">Event Tracking</h2>
+    <p>When the banner is displayed to the user it will automatically track <code>banner</code> event with <code>action: show</code> and same properties as our old banners.</p>
+    <p>It is possible to disable automatic tracking of the banner by setting <code>tracking: false</code> on the Exp instance.</p>
+    <pre><code>var banner = new Exp({
+  tracking: false      
+})</code></pre>
+    <p>Exp adds two directives for tracking:</p>
+    <p>1. <code>exp-action</code> directive will track <code>action: show</code> event on click.</p>
+    <pre><code>&lt;button exp-action&gt;Click me&lt;/button&gt;</code></pre>
+    <p>2. <code>exp-close</code> directive tracks <code>action: close</code> event and removes the banner.</p>
+    <pre><code>&lt;span exp-close&gt;X&lt;/span&gt;</code></pre>
+    <h2 id="sentry-integration">Sentry Integration</h2>
+    <p>Exp is fully integrated with <a href="https://sentry.io/" target="_blank">Sentry.io</a>. Sentry is an open-source error tracking tool that helps developers monitor and fix crashes in real-time. </p>
+    <p>By default Sentry is turned off. Sentry can be enabled on Exp instance:</p>
+    <pre><code>var banner = new Exp({
+  context: this,
+  sentry: {
```

Comment:
@hackal update with current required settings

Reply:
Should be updated